zircon.core/src/main/kotlin/org/hexworks/zircon/internal/component/impl/log/DefaultLogArea.kt

```diff
@@ -0,0 +1,133 @@
+
+    private var logElementBuffer = LogElementBuffer(visibleSize, this, logRowHistorySize)
+
+    init {
+        render()
+    }
+
+    override fun addTextElement(text: String, modifiers: Set<Modifier>?) {
+        val position = getNewElementPositionX()
+        val textElement = LogTextElement(text, position)
+        if (modifiers != null)
+            textElement.modifiers = modifiers
+        logElementBuffer.addLogElement(textElement)
+        render()
+        determineActualSizeAndOffset()
+    }
+
+    override fun addComponentElement(component: Component) {
+        require(component.size.yLength == 1) { "only components with height of 1 are supported" }
+
+        val position = getNewElementPositionX()
+        logElementBuffer.addLogElement(LogComponentElement(component, position))
+        addComponent(component)
+        render()
+        determineActualSizeAndOffset()
+    }
+
+
+    private fun getNewElementPositionX(): Int {
+        val lastElementInRow = logElementBuffer.currentLogElementRow().logElements.lastOrNull()
+        return if (lastElementInRow != null)
+            lastElementInRow.getPosition().x + lastElementInRow.getSize().width()
+        else
+            0
+    }
+
+    override fun addNewRows(numberOfRows: Int) {
+        logElementBuffer.addNewRows(numberOfRows)
+        render()
+        determineActualSizeAndOffset()
+    }
+
+    override fun getLogElementBuffer(): LogElementBuffer {
+        return logElementBuffer
+    }
+
+    override fun clear() {
+        logElementBuffer.clear()
+        children.forEach { removeComponent(it) }
+        actualSize = size
+        scrollUpBy(visibleOffset.y)
+    }
+
+    override fun applyColorTheme(colorTheme: ColorTheme): ComponentStyleSet {
+        return ComponentStyleSetBuilder.newBuilder()
+                .defaultStyle(StyleSetBuilder.newBuilder()
+                        .foregroundColor(colorTheme.secondaryForegroundColor)
+                        .backgroundColor(colorTheme.primaryBackgroundColor)
+                        .build())
+                .disabledStyle(StyleSetBuilder.newBuilder()
+                        .foregroundColor(colorTheme.secondaryForegroundColor)
+                        .backgroundColor(colorTheme.secondaryBackgroundColor)
+                        .build())
+                .focusedStyle(StyleSetBuilder.newBuilder()
+                        .foregroundColor(colorTheme.primaryBackgroundColor)
+                        .backgroundColor(colorTheme.primaryForegroundColor)
+                        .build())
+                .build().also { css ->
+                    componentStyleSet = css
+                    render()
+                    children.forEach {
+                        it.applyColorTheme(colorTheme)
+                    }
+                }
+    }
+
+    override fun render() {
+        renderingStrategy.render(this, tileGraphics)
+    }
+
+    private fun determineActualSizeAndOffset() {
```

Comment:
When I read this I thought that this would return a value. Instead of `determine` `update` might be a better name, what do you think?